ci/test-flow-visibility.sh

```diff
@@ -0,0 +1,156 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+CLICKHOUSE_URL="tcp://clickhouse-clickhouse.flow-visibility:9000"
+FAILURE=false
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+GIT_CHECKOUT_DIR=${THIS_DIR}/..
+
+_usage="Usage: $0 [--kubeconfig <KubeconfigSavePath>]
+
+Setup Grafana, ClickHouse and Antrea Flow Aggregator to validate flow-aggregator.yml and flow-visibility.yml.
+
+    --kubeconfig    Path to kubeconfig.
+
+"
+
+echoerr() {
+  >&2 echo "$@"
+}
+
+print_usage() {
+  echoerr "$_usage"
+}
+
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+case $key in
+    --kubeconfig)
+    export KUBECONFIG="$2"
+    echo "kube"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+start_antrea() {
+  echo "=== Starting Antrea ==="
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/antrea.yml
+  kubectl rollout status --timeout=5m deployment/coredns -n kube-system
+  kubectl rollout status --timeout=5m deployment.apps/antrea-controller -n kube-system
+  kubectl rollout status --timeout=5m daemonset/antrea-agent -n kube-system
+}
+
+setup_flow_visibility() {
+  echo "=== Starting Flow Visibility ==="
+  # install ClickHouse operator
+  kubectl apply -f https://raw.githubusercontent.com/Altinity/clickhouse-operator/master/deploy/operator/clickhouse-operator-install-bundle.yaml
+  kubectl apply -f ${GIT_CHECKOUT_DIR}/build/yamls/flow-visibility.yml
+  echo "=== Waiting for ClickHouse and Grafana to be ready ==="
+  sleep 10
+  kubectl wait --for=condition=ready pod -l app=clickhouse-operator -n kube-system --timeout=600s
+  kubectl wait --for=condition=ready pod -l app=clickhouse -n flow-visibility --timeout=600s
+  kubectl wait --for=condition=ready pod -l app=grafana -n flow-visibility --timeout=600s
+}
+
+check_grafana() {
+  echo "=== Check the installation of Grafana plugins ==="
+  grafana_logs=$(kubectl logs -n flow-visibility $(kubectl -n flow-visibility get pod -l app=grafana -o jsonpath="{.items[0].metadata.name}"))
+  if ( echo ${grafana_logs} | grep -q 'msg="Plugin registered" logger=plugin.manager pluginId=antreaflowvisibility-grafana-sankey-plugin' ); then
+    echo "=== antreaflowvisibility-grafana-sankey-plugin installed correctly ==="
+  else
+    echo "=== antreaflowvisibility-grafana-sankey-plugin is NOT installed correctly ==="
+    FAILURE=true
+  fi
+  if ( echo ${grafana_logs} | grep -q 'msg="Plugin registered" logger=plugin.manager pluginId=grafana-clickhouse-datasource' ); then
+    echo "=== grafana-clickhouse-datasource installed correctly ==="
+  else
+    echo "=== grafana-clickhouse-datasource is NOT installed correctly ==="
+    FAILURE=true
+  fi
+}
+
+check_clickhouse_monitor() {
+  echo "=== Check the status of ClickHouse monitor cron job status==="
+  sleep 60
```

Comment:
@yanjunz97 - do you think we can set a 120sec timeout at line 102 instead of adding this sleep? I'm thinking maybe it's not worth always waiting 60 seconds if the job actually already completed.

Reply:
Cronjob does not launch a job immediately when it is deployed. For example, if the Cronjob is deployed at 12:00:01, it will wait till 12:01:00 to launch its first job. This is why we need to wait for 60s for the first job to be launched. Otherwise `kubectl get job -l app=clickhouse-monitor -n flow-visibility` may not find the job.

I have just opened PR #3498 to run Clickhouse monitor in a long-running pod. If this one is merged, we can get rid of the waiting, and update the check as the other components with something like
``kubectl wait --for=condition=ready pod -l app=clickhouse-monitor -n flow-visibility --timeout=600s``

Reply:
@yanjunz97 so the problem is that maybe we will not have a job at all for which we should verify completion if we don't wait those initial 60 seconds? I'll take a look at the new PR as well.